src/quartz/simulator/schedule.cpp

```diff
@@ -1074,14 +1074,22 @@ get_schedules(const CircuitSeq &sequence,
             }
           }
           bool executed_any_gate = false;
-          for (int j = 0; j < (int)single_qubit_gate_to_execute.size(); j++) {
-            int index = single_qubit_gate_to_execute[j];
+          for (int k = 0; k < (int)single_qubit_gate_to_execute.size(); k++) {
+            int index = single_qubit_gate_to_execute[k];
             bool executable = true;
             for (auto &wire : sequence.gates[index]->input_wires) {
-              if (wire->is_qubit() && schedule.is_local_qubit(wire->index) &&
-                  !current_kernel_qubits[wire->index]) {
-                executable = false; // not local in this kernel
-                break;
+              if (wire->is_qubit()) {
+                if (schedule.is_local_qubit(wire->index)) {
+                  if (!current_kernel_qubits[wire->index]) {
+                    executable = false; // not local in this kernel
+                    break;
+                  }
+                } else {
+                  if (!sequence.gates[index]->gate->is_sparse()) {
```

Comment:
We can put these (for example global X) gates in a separate kernel later.

Reply:
Looks good!